lib/AST/ASTContext.cpp

```diff
@@ -7739,6 +7740,11 @@ QualType ASTContext::mergeFunctionTypes(QualType lhs, QualType rhs,
       if (paramTy->isPromotableIntegerType() ||
           getCanonicalType(paramTy).getUnqualifiedType() == FloatTy)
         return QualType();
+
+      // For Checked C, a no prototype function is not compatible
+      // with a prototype with a checked argument.
+      if (isCheckedC && isNotAllowedForNoProtoTypeFunction(paramTy))
```

Comment:
I might be missing something because I read through this too quickly, but when does this code path gets hit?  Because it's under an `if (proto)`, does that mean it's called in cases when we have a prototype?  If that's the case, I'm not sure I understand why we'd call `isNotAllowedForNoProtoTypeFunction` in cases when we have a yes-has-prototype function.

What am I missing?